src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,108 @@
+package frc.robot.subsystems.conveyor;
+
+import com.ctre.phoenix.motorcontrol.TalonFXControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.utils.DeadbandLaserSensor;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static frc.robot.Constants.Conveyor.*;
+
+public class Conveyor extends SubsystemBase {
+    private static final AtomicInteger balls = new AtomicInteger(Constants.Conveyor.INITIAL_BALLS_AMOUNT);
+
+    private final TalonFX motor = new TalonFX(Ports.Conveyor.MOTOR);
+    private final DeadbandLaserSensor shooterSensor = new DeadbandLaserSensor(Ports.Conveyor.SHOOTER_LASER_SENSOR,
+            SHOOTER_PROXIMITY_LOST_VOLTAGE, SHOOTER_PROXIMITY_SENSE_VOLTAGE);
+    private final DeadbandLaserSensor funnelSensor = new DeadbandLaserSensor(Ports.Conveyor.FUNNEL_LASER_SENSOR,
+            FUNNEL_PROXIMITY_LOST_VOLTAGE, FUNNEL_PROXIMITY_SENSE_VOLTAGE);
+
+    public Conveyor() {
+        motor.setInverted(Ports.Conveyor.IS_MOTOR_INVERTED);
+
+        motor.configPeakOutputForward(FORWARD_PEAK, Constants.TALON_TIMEOUT);
+        motor.configPeakOutputReverse(REVERSE_PEAK, Constants.TALON_TIMEOUT);
+
+        motor.configSupplyCurrentLimit(SUPPLY_CURRENT_LIMIT, Constants.TALON_TIMEOUT);
+        motor.configStatorCurrentLimit(STATOR_CURRENT_LIMIT, Constants.TALON_TIMEOUT);
+
+        motor.enableVoltageCompensation(true);
+        motor.configVoltageCompSaturation(Constants.NOMINAL_VOLTAGE);
+    }
+
+    /**
+     * Get the amount of balls sensed by the proximity.
+     *
+     * @return the amount of balls in the conveyor.
+     */
+    private static int getBallsAmount() {
+        return balls.get();
+    }
+
+    public static synchronized boolean isConveyorFull() {
+        return getBallsAmount() >= MAX_BALLS_AMOUNT;
+    }
+
+    public static synchronized boolean isConveyorEmpty() {
+        return getBallsAmount() <= 0;
+    }
+
+    /**
+     * Increment by one the amount of balls (only programmatically).
+     */
+    private static void addBall() {
+        balls.incrementAndGet();
+    }
+
+    /**
+     * Decrement by one the amount of balls (only programmatically).
+     */
+    private static void removeBall() {
+        balls.decrementAndGet();
+    }
+
+    /**
+     * Get whether the power applied by the motor is upward, i.e moving up.
+     *
+     * @return whether the power applied by the motor is upward.
+     */
+    private boolean isMovingUp() {
+        return motor.getMotorOutputPercent() > 0;
```

Comment:
Because it is talonFX and you have a built in encoder it would be more accurate if you check the speed and not the precent output

Reply:
You don't really need accuracy, you just want to know the direction the conveyor is moving in.
However, maybe you should have another method that returns whether the conveyor is moving down because this method returns `false` if the conveyor doesn't move at all.

Reply:
I don't agree with you @katzuv, if we are not moving up, the balls still can slip and it won't handle it

Reply:
> it won't handle it

I'm not sure what you mean, please be more specific.
> if we are not moving up, the balls still can slip

If Power Cells slip when the conveyor is static, they will slip when it moves, too. The friction between Power Cells and the conveyor is the same, regardless of whether it's moving or not.

Reply:
> The friction between Power Cells and the conveyor is the same, regardless of whether it's moving or not.

?

> I'm not sure what you mean, please be more specific.

I'm talking about the case when the motor doesn't apply power and the balls slip

Reply:
The force that drives the Power Cells up in the conveyor is static friction. IIRC from my physics classes, the formula for static friction between two objects is <img src="https://render.githubusercontent.com/render/math?math=F_s = \mu_s N">. This force's magnitude is not dependant on the velocity of any of the two objects.
Thus, if Power Cells slip when the conveyor is static, they'll slip when it moves, too.

Also, I'm not sure how we came to discuss this, because my original comment was:
> maybe you should have another method that returns whether the conveyor is moving down because this method returns `false` if the conveyor doesn't move at all.
https://github.com/Galaxia5987/robot-2021/pull/9#discussion_r567111457